### PR TITLE
Improve the health description for Cloud Run live state

### PR DESCRIPTION
**Pull request by knanao**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/platformprovider/cloudrun/cloudrun.go [hunk 1]
```diff
@@ -192,23 +238,60 @@ func (r *Revision) RevisionManifest() (RevisionManifest, error) {
...
-	conds := status.Conditions
-	for _, c := range conds {
-		isHealthy, err := strconv.ParseBool(c.Status)
-		if err != nil {
-			return model.CloudRunResourceState_UNKNOWN, fmt.Sprintf(errMessage, "unable to parse status: %s", c.Status)
-		}
-		if !isHealthy {
-			return model.CloudRunResourceState_OTHER, c.Message
+	if len(s.FalseMessages) > 0 {
+		return model.CloudRunResourceState_OTHER, fmt.Sprintf(errMessage, strings.Join(s.FalseMessages, ""))
```
Comment (nghialv) [inline pkg/app/piped/platformprovider/cloudrun/cloudrun.go:280]: This should not be a kind of `Unpextecd error while calculating:` .

File: pkg/app/piped/platformprovider/cloudrun/cloudrun.go [hunk 2]
```diff
@@ -192,23 +238,60 @@ func (r *Revision) RevisionManifest() (RevisionManifest, error) {
...
+	}
+
+	if len(s.UnknownMessages) > 0 {
+		return model.CloudRunResourceState_UNKNOWN, fmt.Sprintf(errMessage, strings.Join(s.UnknownMessages, ""))
```
Comment (nghialv) [inline pkg/app/piped/platformprovider/cloudrun/cloudrun.go:284]: This should not be a kind of `Unpextecd error while calculating:` .

File: pkg/app/piped/platformprovider/cloudrun/cloudrun.go [hunk 3]
```diff
@@ -192,23 +238,60 @@ func (r *Revision) RevisionManifest() (RevisionManifest, error) {
...
+	}
+
+	mustPassConditions := TypeHelthyServiceConditions
+	if s.Kind == KindRevision {
+		mustPassConditions = TypeHelthyRevisionConditions
+	}
+	for k := range mustPassConditions {
+		if _, ok := s.TrueMessages[k]; !ok {
+			errMsg := fmt.Sprintf("could not check status field %q", k)
+			return model.CloudRunResourceState_UNKNOWN, fmt.Sprintf(errMessage, errMsg)
```
Comment (nghialv) [inline pkg/app/piped/platformprovider/cloudrun/cloudrun.go:294]: This should not be a kind of `Unpextecd error while calculating:` .

File: pkg/app/piped/platformprovider/cloudrun/cloudrun.go [hunk 4]
```diff
@@ -40,9 +39,46 @@ var (
...
+var (
+	TypeConditions = map[string]struct{}{
+		"Active":              struct{}{},
+		"Ready":               struct{}{},
+		"ConfigurationsReady": struct{}{},
+		"RoutesReady":         struct{}{},
+		"ContainerHealthy":    struct{}{},
+		"ResourcesAvailable":  struct{}{},
+	}
+	TypeHelthyServiceConditions = map[string]struct{}{
```
Comment (nghialv) [inline pkg/app/piped/platformprovider/cloudrun/cloudrun.go:51]:
```suggestion
	TypeHealthyServiceConditions = map[string]struct{}{
```

**Comment by knanao** [comment]
@nghialv 
I updated some points, so please check it out when you have time.

**Review by nghialv — APPROVED** [review]
Looks nice.

**Pull request merged**